### PR TITLE
cmake: Use cmake var not env for sdk

### PR DESCRIPTION
**Pull request by jeremybettis**
ZEPHYR_SDK_INSTALL_DIR is set from the env if it is not passed
explicitly to cmake, but one usage was reading directly from the env,
not allowing passing the var to cmake instead.

Change find_package of Zephyr-sdk to read from cmake var not env var.

Signed-off-by: Jeremy Bettis <jbettis@google.com>
Change-Id: I778fe7d325852890148436adc4aa59ed686602c3

### CODE REVIEW

**Review by aaronemassey — DISMISSED** [review]

**Review by alevkoy — DISMISSED** [review]

**Review by aaronemassey — APPROVED** [review]

**Review by yperess — CHANGES_REQUESTED** [review]
You can list both as hints, I think we should keep the environment variables for downstream reasons no?

**Review by yperess — APPROVED** [review]
LGTM

**Review by fabiobaltieri — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

File: cmake/modules/verify-toolchain.cmake [hunk 1]
```diff
@@ -76,7 +76,7 @@ if(("zephyr" STREQUAL ${ZEPHYR_TOOLCHAIN_VARIANT}) OR
     # To support Zephyr SDK tools (DTC, and other tools) with 3rd party toolchains
     # then we keep track of current toolchain variant.
     set(ZEPHYR_CURRENT_TOOLCHAIN_VARIANT ${ZEPHYR_TOOLCHAIN_VARIANT})
-    find_package(Zephyr-sdk ${TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION} REQUIRED QUIET HINTS $ENV{ZEPHYR_SDK_INSTALL_DIR})
+    find_package(Zephyr-sdk ${TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION} REQUIRED QUIET HINTS ${ZEPHYR_SDK_INSTALL_DIR})
```
Comment (tejlmand) [inline cmake/modules/verify-toolchain.cmake:79]: no, this should not change.

The environment is how users are used to work with ZEPHYR_SDK_INSTALL_DIR, CMake itself offers the possibility of using `<packagename>_ROOT` for seaching, which can be set as CMake variable.

Ref: https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-search-procedure
> 1. New in version 3.12: Search paths specified in the `<PackageName>_ROOT` CMake variable and the `<PackageName>_ROOT` environment variable, where <PackageName> is the package to be found.

**Review by tejlmand — CHANGES_REQUESTED** [review]

File: cmake/modules/verify-toolchain.cmake (hunk 1 above)
Comment (tejlmand) [reply cmake/modules/verify-toolchain.cmake:79]: My bad, I didn't notice we are already mapping the environment setting into a CMake variable (if not already set):
https://github.com/zephyrproject-rtos/zephyr/blob/cd7c44a152bc4469ad7da9e38d58df46655d460d/cmake/modules/verify-toolchain.cmake#L28-L29

so agreed, better to use the CMake var on this.
Although we should probably consider better alignment with CMake natively supported `find_package()`, but that is outside scope of this PR.

**Review by tejlmand — APPROVED** [review]
lgtm: https://github.com/zephyrproject-rtos/zephyr/pull/48717#discussion_r938742513

**Pull request merged**